ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/ViewVulnReport/RunHistory.tsx

```diff
@@ -0,0 +1,159 @@
+import React, { ReactElement } from 'react';
+import { TableComposable, Tbody, Td, Th, Thead, Tr } from '@patternfly/react-table';
+import { Bullseye, Card, Divider, Spinner } from '@patternfly/react-core';
+
+import { ReportConfiguration, ReportRequestType } from 'services/ReportsService.types';
+import { SlimUser } from 'types/user.proto';
+import { getDateTime } from 'utils/dateUtils';
+import { getReportFormValuesFromConfiguration } from 'Containers/Vulnerabilities/VulnerablityReporting/utils';
+import useSet from 'hooks/useSet';
+
+import NotFoundMessage from 'Components/NotFoundMessage/NotFoundMessage';
+import EmptyStateTemplate from 'Components/PatternFly/EmptyStateTemplate/EmptyStateTemplate';
+import { CubesIcon } from '@patternfly/react-icons';
+import LastRunStatusState from '../VulnReports/LastRunStatusState';
+import ReportParametersDetails from '../components/ReportParametersDetails';
+import DeliveryDestinationsDetails from '../components/DeliveryDestinationsDetails';
+import ScheduleDetails from '../components/ScheduleDetails';
+import useFetchReportHistory from '../api/useFetchReportHistory';
+
+export type RunTypeProps = {
+    reportRequestType: ReportRequestType;
+    user: SlimUser;
+};
+
+function RunType({ reportRequestType, user }: RunTypeProps): ReactElement {
+    if (reportRequestType === 'ON_DEMAND') {
+        return (
+            <div>
+                On-demand / <span className="pf-u-color-200">Requested by {user.name}</span>
+            </div>
+        );
+    }
+    if (reportRequestType === 'SCHEDULED') {
+        return <div>Scheduled</div>;
+    }
+    return <div>-</div>;
+}
+
+export type RunHistoryProps = {
+    reportId: string;
+};
+
+function RunHistory({ reportId }: RunHistoryProps) {
+    const { reportSnapshots, isLoading, error } = useFetchReportHistory({
+        id: reportId,
+    });
+    const expandedRowSet = useSet<string>();
+
+    if (isLoading) {
+        return (
+            <Bullseye>
+                <Spinner isSVG />
+            </Bullseye>
+        );
+    }
+
+    if (error) {
+        return (
+            <NotFoundMessage
+                title="Error fetching report history"
+                message={error || 'No data available'}
+            />
+        );
+    }
+
+    if (!reportSnapshots.length) {
+        return (
+            <Bullseye>
+                <EmptyStateTemplate title="No run history" headingLevel="h2" icon={CubesIcon} />
+            </Bullseye>
+        );
+    }
+
+    return (
+        <TableComposable aria-label="Simple table" variant="compact">
+            <Thead>
+                <Tr>
+                    <Th />
```

Comment:
Minor accessibility nit I've had on some of my recent PRs. The aXe devtool will report an error for an empty `<Th>` element in a table header. You can safely replace this with an empty `<Td />` to avoid the error.